### PR TITLE
Qr page update

### PR DESCRIPTION
**Pull request by nexticodes**
[//]: # (Title Template: "[TR_42] QR Page update [DONOR] + Confirmation Popup")
---

#### Please check if the PR fulfills these requirements

- [x] The changes don't come from your master branch.

---

#### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
This change introduces the following: 
1) Added yellow reticles and change the text color from blue to white with shadows on BarCodeMask component.
2) Popup modal confirmation appears upon scanning a QR code or when an error appears.

#### What is the current behavior? (You can also link to an open issue here)
The current QR page (as shown in the photo down below) has some missing icons and other functionalities.  It only brings the user back to the previous page upon scanning -- no modal components showing or indication that it scanned successfully. Additionally, it does not have an "X" on the top-right, chevron-left is a bit too big, blue text displaying instructions might be troublesome in some cases.

#### What is the new behavior? (if this is a feature change)
New behavior includes the modal component popping up when a scan is either successful or a failure. The modal component also displays important information about the scanned QR code. For this case, I only used a static, dummy data to display to the modal (will need to be dynamic, but that's a different PR) but the general format has been implemented.
Additionally, changes to the general layout of the QR Scan page  (inserted a few icons, add yellow reticle, change text color to a more pleasing color with added effects to increase visibility).

#### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)

No breaking changes.

#### Other information

1) I might need to find a way to somehow temporarily disable the camera or stop the camera from scanning upon scan success to avoid potential duplicates when a user decides to not move their camera away from the QR code after scanning (super edge-case).

2) The colors may be off on the demo I presented but that's because I had to compress the original GIF file in order for me to be able to include them here.

Ready to make a change whenever! Thank you all for reviewing :)

---

#### Images (before/ after screenshots, interaction GIFs, ...)
![Before/after example image]

#### BEFORE
Screenshot_1585352849

##### Scan success
Scan success min

##### Scan fail
Scan Fail min

### CODE REVIEW

**Comment by duskcloudxu** [comment]
Good job, but it seems your PR did not pass the lint requirement...maybe you could find the files containing lint problem in build failing details and fix it before merge?

**Comment by duskcloudxu** [comment]
tested your code and there is misalignment in IOS simulator, found possible reason and will comment on corresponding line lol.

File: src/screens/QRCodeScannerScreen/BarCodeMask.styles.ts [hunk 1]
```diff
@@ -2,69 +2,105 @@ import { StyleSheet, Dimensions } from 'react-native';
...
+		paddingHorizontal: 5,
 	},
 	x: {
 		textAlign: 'center',
-		textAlignVertical: 'center',
-		fontSize: 70,
+		fontSize: 48,
 	},
 	textContainer: {
 		position: 'absolute',
```
Comment (duskcloudxu) [inline src/screens/QRCodeScannerScreen/BarCodeMask.styles.ts:39]: I guess it's not good to use absolute position there, it does not compatible with flexbox, and I think that's the reason of misalignment in my IOS simulator....

Comment (nexticodes) [reply src/screens/QRCodeScannerScreen/BarCodeMask.styles.ts:39]: OOPS -- Thanks Tao! I'll look into it :)

File: src/screens/QRCodeScannerScreen/BarCodeMask.styles.ts [hunk 2]
```diff
@@ -36,7 +36,6 @@ export default StyleSheet.create({
 		fontSize: 48,
 	},
 	textContainer: {
-		position: 'absolute',
```
Comment (nexticodes) [inline src/screens/QRCodeScannerScreen/BarCodeMask.styles.ts:20]: Removing this didn't change anything on my end -- the text remained in the same position. Is it the same for you @duskcloudxu ?

**Pull request merged**